src/ophyd_async/tango/demo/counter.py

```diff
@@ -0,0 +1,52 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    AsyncStatus,
+    ConfigSignal,
+    HintedSignal,
+    Signal,
+    SignalX,
+)
+from ophyd_async.tango import (
+    TangoReadableDevice,
+)
+
+
+@dataclass
+class TangoCounterConfig:
+    sample_time: Optional[float] = None
+
+
+class TangoCounter(TangoReadableDevice):
+    # Enter the name and type of the signals you want to use
+    # If type is None or Signal, the type will be inferred from the Tango device
+    counts: None
+    sample_time: Signal
```

Comment:
When we do this in PandA, we specify not just the names of the signals, but their types, e.g.:
```suggestion
    sample_time: SignalRW[float]
```
so that when we `await self.sample_time.get_value()` we know we are getting a float. We then check at connection time that the EPICS PV is actually a float, and error if not. This means we can catch things like `32 + await self.my_string_attribute.get_value()` at lint time rather than runtime

Is it possible to do the same here?

Reply:
Done